### DISCUSSION TITLE
Not able to request the profile list

### DISCUSSION THREAD

**Discussion by zonwizard** [denisneuf/python-amazon-ad-api#63] [Q&A]
In order to get the profile list I need to call the Profile().list_profiles() operation.
But to initialize the Profile client a profile_id is required.
How can i get a profile_id before calling the list_profiles operation?

Thanks

**Comment by denisneuf** [comment]
Check this:
[https://advertising.amazon.com/API/docs/en-us/getting-started/retrieve-profiles](https://advertising.amazon.com/API/docs/en-us/getting-started/retrieve-profiles)

**Comment by zonwizard** [comment]
Thanks @denisneuf ,
from what I read in the docu in order to call the profile endpoint **/v2/profiles**
it should be enough to pass 2 parameters **client_id** and **access_token**
```
curl  \
    -H "Amazon-Advertising-API-ClientId: YOUR_CLIENT_ID"\
    -H "Authorization: Bearer YOUR_ACCESS_TOKEN" \
    https://advertising-api.amazon.com/v2/profiles
```

This operation in this project is the Profile().list_profiles() right?
The problem is that a profile_id is a required parameter to initialize the Profile client when it should not be required to call the list_profiles operation.


To contextualize, I need to access the ad api for sellers that authorize my app and I don't have their profile_id before calling the list_profiles.
What are the operations to get their profile_ids?

Thanks for your help

Guido
